### PR TITLE
`CryptoApproveAllowance` and `CryptoDeleteAllowance`  handle implementation

### LINKED ISSUE #6391: CryptoApproveAllowanceHandler

### LINKED ISSUE #6408: Add `pureChecks` and `validateSemantics` implementations for  CryptoApproveAllowanceHandler

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #6391 
Fixes #6408 
Fixes https://github.com/hashgraph/hedera-services/issues/6392
Fixes https://github.com/hashgraph/hedera-services/issues/6409
- Adds `CryptoApproveAllowance` and `CryptoDeleteAllowance `handle() implementation
- Moved helper and base classes in tests to a different package
- Added `AllowancesValidator` to separate `validateSemantics` logic for approving allowances
- Added `ApproveAllowanceValidator` and `DeleteAllowanceValidator` to validate txn in handle()
- Added `ReadableUniqueTokenStore` and `WritableUniqueTokenStore`

Testing:
Unit tests

### CODE REVIEW

**Review by mhess-swl — DISMISSED** [review]

**Review by tinker-michaelj — DISMISSED** [review]
Very nice work @Neeharika-Sompalli, LGTM! 🥂

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/ReadableNftStore.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Nullable
+    Nft get(@NonNull TokenID id, long serialNumber);
+
+    /**
+     * Gets {@link Nft} data for a given {@link UniqueTokenId}.
+     * @param id the unique token id to look up
+     * @return the {@link Nft} data for the given unique token id, {@code null} if the id doesn't exist
+     */
+    @Nullable
+    Nft get(@NonNull UniqueTokenId id);
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/ReadableNftStore.java:46]: Yes. That makes sense. Will do

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableNftStore.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     * Persists a new {@link Nft} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param nft - the nft to be persisted.
+     */
+    public void put(@NonNull final Nft nft) {
+        Objects.requireNonNull(nft);
+        nftState.put(nft.id(), Objects.requireNonNull(nft));
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableNftStore.java:61]: Removed. Thanks !

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableNftStore.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * Commits the changes to the underlying data storage.
+     */
+    public void commit() {
+        ((WritableKVStateBase) nftState).commit();
+    }
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableNftStore.java:69]: Got it ! I wasn't sure either if it should exist here. Removed from all writable stores.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/ContextualRetriever.java [hunk 4]
```diff
@@ -58,4 +59,8 @@ public static Token getIfUsable(@NonNull final TokenID tokenId, @NonNull final R
         validateFalse(token.paused(), TOKEN_IS_PAUSED);
         return token;
     }
+
+    public static boolean isFungibleCommon(@NonNull final Token token) {
+        return token.tokenType().equals(TokenType.FUNGIBLE_COMMON);
+    }
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/ContextualRetriever.java:65]: Yeah 😄 . It is basically a helper class with this `getIfUsable` method that is used in many token handlers. I will probably rename to TokenHandlerHelper. `isFungibleCommon` is mostly used in customFees, approve allowance and delete allowance. I can delete it

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/ReadableNftStore.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/ReadableNftStore.java]: Will change `UniqueTokenId` in the next PR

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableNftStoreImpl.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Returns the NFT leaf for the given nftId. If the token and its serial doesn't exist returns {@code
+     * null}
+     *
+     * @param nftId given token Id and its serial number
+     * @return Nft leaf for the given unique token Id
+     */
+    private Nft getNftLeaf(final UniqueTokenId nftId) {
+        return nftState.get(nftId);
+    }
```
Comment (rbair23) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableNftStoreImpl.java:63]:
```suggestion
    @Override
    @Nullable
    public Nft get(@NonNull final UniqueTokenId id) {
        requireNonNull(id);
        return nftState.get(nftId);
    }
```

**Review by rbair23 — DISMISSED** [review]

**Pull request merged**